Fix from upstream for running tests when LC_ALL is set to non-C

http://bugs.gentoo.org/149147

Index: make/tests/run_make_tests.pl
===================================================================
RCS file: /cvsroot/make/make/tests/run_make_tests.pl,v
retrieving revision 1.22
retrieving revision 1.23
diff -u -p -r1.22 -r1.23
--- make/tests/run_make_tests.pl	8 Mar 2006 20:15:09 -0000	1.22
+++ make/tests/run_make_tests.pl	1 Oct 2006 05:38:38 -0000	1.23
@@ -228,11 +228,6 @@ sub set_more_defaults
    local($string);
    local($index);
 
-   # Make sure we're in the C locale for those systems that support it,
-   # so sorting, etc. is predictable.
-   #
-   $ENV{LANG} = 'C';
-
    # find the type of the port.  We do this up front to have a single
    # point of change if it needs to be tweaked.
    #
Index: make/tests/test_driver.pl
===================================================================
RCS file: /cvsroot/make/make/tests/test_driver.pl,v
retrieving revision 1.19
retrieving revision 1.20
diff -u -p -r1.19 -r1.20
--- make/tests/test_driver.pl	10 Mar 2006 02:20:45 -0000	1.19
+++ make/tests/test_driver.pl	1 Oct 2006 05:38:38 -0000	1.20
@@ -78,9 +78,9 @@ sub resetENV
 sub toplevel
 {
   # Pull in benign variables from the user's environment
-  #
+
   foreach (# UNIX-specific things
-           'TZ', 'LANG', 'TMPDIR', 'HOME', 'USER', 'LOGNAME', 'PATH',
+           'TZ', 'TMPDIR', 'HOME', 'USER', 'LOGNAME', 'PATH',
            # Purify things
            'PURIFYOPTIONS',
            # Windows NT-specific stuff
@@ -92,6 +92,10 @@ sub toplevel
     $makeENV{$_} = $ENV{$_} if $ENV{$_};
   }
 
+  # Make sure our compares are not foiled by locale differences
+
+  $makeENV{LC_ALL} = 'C';
+
   # Replace the environment with the new one
   #
   %origENV = %ENV;
